### PR TITLE
Issue 35 exception handler in every block

### PR DESCRIPTION
**Pull request by mzabolotko**
Resolved issue https://github.com/mzabolotko/Kontur/issues/35

### CODE REVIEW

File: Sources/Kontur/MessageSubscriber.cs [hunk 1]
```diff
@@ -27,6 +31,17 @@ public ISubscriptionTag SubscribeTo(ISourceBlock<IMessage> target)
...
         }
 
+        private void OnMessage(IMessage message)
+        {
+            try
+            {
+                action(this.As(message));            
+            }
+            catch (System.Exception)
+            {
```
Comment (alexzaitzev) [inline Sources/Kontur/MessageSubscriber.cs:41]: Empty catch? What should be done here?

File: Sources/Kontur.Rabbitmq/AmqpSender.cs [hunk 2]
```diff
@@ -23,28 +24,50 @@ public ISubscriptionTag SubscribeTo(ISourceBlock<IMessage> source)
...
+                    }
+                    catch (Exception ex)
+                    {
+                        return new Result<AmqpMessage>(ExceptionDispatchInfo.Capture(ex));
+                    }}));
 
-            var amqpSenderBlock = new ActionBlock<AmqpMessage>(
-                    (Action<AmqpMessage>)this.Send);
+            var amqpSenderBlock = new ActionBlock<Result<AmqpMessage>>(
+                    (Action<Result<AmqpMessage>>)this.Send);
```
Comment (alexzaitzev) [inline Sources/Kontur.Rabbitmq/AmqpSender.cs]: For what `Result<>` was added?

**Review by alexzaitzev — CHANGES_REQUESTED** [review]
Need to answer on questions in comments

File: Sources/Kontur.Rabbitmq/AsyncAmqpBasicConsumer.cs [hunk 3]
```diff
@@ -32,47 +39,78 @@ class AsyncAmqpBasicConsumer<T> : IPublisher where T : class
...
+                (Func<MessageResult, IMessage>)(result => result.Value));
+            this.handleErrorBlock = new ActionBlock<MessageResult>(
+                ((Action<MessageResult>)(result =>
+                    {
+                        try
+                        {
+                            this.channel.BasicNack(result.Error.Delivery.DeliveryTag, false, false);
+                        }
+                        catch (Exception)
+                        {
```
Comment (alexzaitzev) [inline Sources/Kontur.Rabbitmq/AsyncAmqpBasicConsumer.cs:76]: Empty catch?

**Review by alexzaitzev — CHANGES_REQUESTED** [review]

File: Sources/Kontur/MessageSubscriber.cs (hunk 1 above)
Comment (mzabolotko) [reply Sources/Kontur/MessageSubscriber.cs:41]: I think this is a place of the implementation of https://github.com/mzabolotko/Kontur/issues/36 and https://github.com/mzabolotko/Kontur/issues/38

File: Sources/Kontur.Rabbitmq/AsyncAmqpBasicConsumer.cs (hunk 3 above)
Comment (mzabolotko) [reply Sources/Kontur.Rabbitmq/AsyncAmqpBasicConsumer.cs:76]: I created issue https://github.com/mzabolotko/Kontur/issues/48 to cover the common case of channel exception handling. This catch will be redisigned during its implementation.

**Review by alexzaitzev — APPROVED** [review]

**Pull request merged**